content/en/blog/_posts/2024-03-07-cri-o-seccomp-oci-artifacts.md

```diff
@@ -0,0 +1,485 @@
+running as seccomp `Unconfined`:
+
+```shell
+cat pod.yaml
+```
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+      securityContext:
+        seccompProfile:
+          type: Unconfined
+```
+
+```shell
+kubectl apply -f pod.yaml
+```
+
+The workload is up and running:
+
+```shell
+kubectl get pods
+```
+
+```console
+NAME   READY   STATUS    RESTARTS   AGE
+pod    1/1     Running   0          15s
+```
+
+And no seccomp profile got applied if I inspect the container using
+[`crictl`](https://sigs.k8s.io/cri-tools):
+
+```shell
+export CONTAINER_ID=$(sudo crictl ps --name container -q)
+sudo crictl inspect $CONTAINER_ID | jq .info.runtimeSpec.linux.seccomp
+```
+
+```console
+null
+```
+
+Now, let's modify the pod to apply the profile `quay.io/crio/seccomp:v2` to the
+container:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    seccomp-profile.kubernetes.cri-o.io/container: quay.io/crio/seccomp:v2
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+```
+
+I have to delete and recreate the Pod, because only recreation will apply a new
+seccomp profile:
+
+```shell
+kubectl delete pod/pod
+```
+
+```console
+pod "pod" deleted
+```
+
+```shell
+kubectl apply -f pod.yaml
+```
+
+```console
+pod/pod created
+```
+
+The CRI-O logs will now indicate that the runtime pulled the artifact:
+
+```console
+WARN[…] Allowed annotations are specified for workload [seccomp-profile.kubernetes.cri-o.io]
+INFO[…] Found container specific seccomp profile annotation: seccomp-profile.kubernetes.cri-o.io/container=quay.io/crio/seccomp:v2  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Pulling OCI artifact from ref: quay.io/crio/seccomp:v2  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+INFO[…] Retrieved OCI artifact seccomp profile of len: 18853  id=26ddcbe6-6efe-414a-88fd-b1ca91979e93 name=/runtime.v1.RuntimeService/CreateContainer
+```
+
+And the container is finally using the profile:
+
+```shell
+export CONTAINER_ID=$(sudo crictl ps --name container -q)
+sudo crictl inspect $CONTAINER_ID | jq .info.runtimeSpec.linux.seccomp | head
+```
+
+```yaml
+{
+  "defaultAction": "SCMP_ACT_ERRNO",
+  "defaultErrnoRet": 38,
+  "architectures": [
+    "SCMP_ARCH_X86_64",
+    "SCMP_ARCH_X86",
+    "SCMP_ARCH_X32"
+  ],
+  "syscalls": [
+    {
+```
+
+The same would work for every container in the pod, if users replace the
+`/container` suffix with the reserved name `/POD`, for example:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    seccomp-profile.kubernetes.cri-o.io/POD: quay.io/crio/seccomp:v2
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+```
+
+{{< note >}}
+CRI-O has an option called `seccomp_use_default_when_empty` (config file) /
```

Comment:
this option has been deprecated and also doesn't work for a while. we can drop it and then could drop this note for 1.30?